File: Beecrowd/1118 Várias Notas Com Validação/main.py
def entradaValidacao():
    val = -1
    while val < 0 or val > 10:
        val = float(input())
        if val < 0 or val > 10:
            print("nota invalida")
    return val


while True:

    x = entradaValidacao()
    y = entradaValidacao()

    media = (x + y) / 2
    print("media = %.2f" % media)

    opc = 0
    while opc != 1 and opc != 2:
        print("novo calculo (1-sim 2-nao)")
        opc = int(input())

    if opc == 2:
        break

File: Beecrowd/1118 Várias Notas Com Validação/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);

        double x, y, media;
        int opc;

        while (true) {

            x = entradaValidacao(sc);
            y = entradaValidacao(sc);

            media = (x + y) / 2;

            System.out.printf("media = %.2f\n", media);

            do {
                System.out.println("novo calculo (1-sim 2-nao)");
                opc = sc.nextInt();
            } while (opc != 1 && opc != 2);

            if (opc == 2)
                break;
        }

        sc.close();
    }

    public static double entradaValidacao(Scanner sc) {
        double val;

        do {
            val = sc.nextDouble();
            if (val < 0.0 || val > 10.0)
                System.out.println("nota invalida");
        } while (val < 0.0 || val > 10.0);

        return val;
    }
}